src/Components/Component.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <queue>
+#include <iostream>
+using namespace std;
+template<class queueType>
+class Component {
+public:
+	void receive(queueType input) {
+		if (active) {
```

Comment:
Implementation should be in .cc file. Please create a separate Component.cc and have it in a separate commit/PR.